File: flutter/lib/utilities/phone_formatter.dart
class PhoneFormatter {
  static String normalizeToE164(String phone) {
    var cleaned = phone.replaceAll(RegExp(r'[\s\-\(\)]'), '');
    if (!cleaned.startsWith('+')) {
      cleaned = '+$cleaned';
    }
    return cleaned;
  }

  static bool isValidE164(String phone) {
    return RegExp(r'^\+[1-9]\d{6,14}$').hasMatch(phone);
  }
}

File: android/app/src/main/java/com/otppoc/android/utilities/PhoneFormatter.kt
package com.otppoc.android.utilities

object PhoneFormatter {

    fun normalizeToE164(phone: String): String {
        var cleaned = phone.replace(Regex("[\\s\\-()]"), "")
        if (!cleaned.startsWith("+")) {
            cleaned = "+$cleaned"
        }
        return cleaned
    }

    fun isValidE164(phone: String): Boolean {
        return Regex("^\\+[1-9]\\d{6,14}$").matches(phone)
    }
}
